trl/trainer/ppov2_trainer.py

```diff
@@ -0,0 +1,799 @@
+from accelerate.utils import broadcast, gather_object
+from datasets import Dataset
+from torch.utils.data import DataLoader
+from transformers import (
+    DataCollatorWithPadding,
+    GenerationConfig,
+    PreTrainedTokenizer,
+    Trainer,
+    TrainerCallback,
+    TrainerControl,
+    TrainerState,
+    TrainingArguments,
+)
+from transformers.integrations import get_reporting_integration_callbacks
+from transformers.trainer_callback import CallbackHandler, DefaultFlowCallback
+
+from trl.models.utils import unwrap_model_for_generation
+from trl.trainer.utils import exact_div, print_rich_table
+
+
+INVALID_LOGPROB = 1.0
+
+
+@dataclass
+class PPOConfig(TrainingArguments):
+    # common config
+    exp_name: str = os.path.basename(__file__)[: -len(".py")]
+    """the name of this experiment"""
+    run_name: Optional[str] = None
+    """a unique name of this run"""
+    sanity_check: bool = False
+    """wether to run in debug mode"""
+
+    # various batch sizes
+    world_size: Optional[int] = None
+    """The number of processes (GPUs) to use"""
+    num_updates: Optional[int] = None
+    """The number of updates to train"""
+    total_episodes: Optional[int] = None
+    """The total number of episodes in the dataset"""
+    micro_batch_size: Optional[int] = None
+    """The micro batch size across devices (HF's `per_device_train_batch_size` * `world_size`)"""
+    local_batch_size: Optional[int] = None
+    """The batch size per GPU (HF's `per_device_train_batch_size` * `gradient_accumulation_steps`)"""
+    batch_size: Optional[int] = None
+    """The batch size across devices (HF's `per_device_train_batch_size` * `world_size` * `gradient_accumulation_steps`)"""
+    num_mini_batches: int = 1
+    """Number of minibatches to split a batch into"""
+    local_mini_batch_size: Optional[int] = None
+    """the mini batch size per GPU"""
+    mini_batch_size: Optional[int] = None
+    """the mini batch size across GPUs"""
+    local_eval_batch_size: int = 2
+    """per rank eval batch size"""
+    local_rollout_forward_batch_size: int = 64
+    """per rank no grad forward pass in the rollout phase"""
+    num_sample_generations: int = 10
+    """the number of debugging samples generations (i.e., `generate_completions` calls) throughout training"""
+
+    # other config
+    base_model: str = "EleutherAI/pythia-160m"
+    """the name of the pretrained model to use"""
+    response_length: int = 53
+    """the length of the response"""
+    truncate_token: Optional[Literal["eos"]] = None
+    """the truncate token"""
+    truncate_token_id: Optional[int] = None
+    """the truncation token id"""
+    temperature: float = 0.7
+    """the sampling temperature"""
+    penalty_reward_value: int = -1
+    """the reward value for responses that do not contain `truncate_token_id`"""
+    non_eos_penalty: bool = False
+    """whether to penalize responses that do not contain `truncate_token_id`"""
+    offload: bool = False
```

Comment:
Sorry if I mentioned this before, but this can be handled automatically by `accelerate` right? I.e. the user just passes `accelerate launch --config_file=config_with_offload.yaml` and then we handle the offloading internally

Reply:
ah sorry forgot to remove it. I just removed it and in practice DS3 would just work.